docs/contributor/performance-tests/metrics/metrics.md

```diff
@@ -0,0 +1,690 @@
+```
+
+## Summary
+
+Overall test result following findings
+
+Setup and parameters
+- Used metric size was `20` metric data points with `10` labels per each distinct metric
+- `1Gi` memory and `1` CPU
+- `memory_limiter` configured for `85%` memory limit, `10%` spike limit with `0.5` second check interval which result (85 -10) `75%` hard limit equivalent to `750Mi` memory.
+- Batch processor configured to create batches with `1024` metrics, default batch size `8192` was exceeded grpc client default payload limit `4MB`
+- Metric gateway configured with a log exporter to avoid possible outages by the gateway back pressure (default 2 gateway instances are available) 
+
+
+Findings:
+- Test results with a single agent instance and single receiver configuration, a single target scrape reach max `~120K` metric data point.
+- Single agent instance with single receiver reached `~280K` metric points per scrape (`~14K metric point/sec`) and stay stable, anything above result metric data refused by the `memory_limiter` processor.
+- Multi node test with single receiver performed on 15 Nodes cluster `1430` pods (this was the max pod count can be deployed on this cluster), agent was able to scrape in total `~2.800K` metrics data per scrape loop and was pushed successfully to the agent without any data dropping by agent side.
+  - No Memory issues identified and no additional memory impact found
+  - K8s Api server request duration reach for `GET` operations to the `~750ms` and for `LIST` operations `~900ms`, there was only one short peak of `21s` detected for `GET` operation
+- `sample_limit` configuration tested for multiple receivers, it's work but there are no metrics identified to see limit violations when occurs, only logs are present in this case
+- Multi receiver test reach same stable limit of `~280K` metrics per scrape in total but anything over this limit can cause OOM, OOM observed only when test execution with high load run over 2 hours. Further investigation and analysis required here to improve memory setting to get more resilience. 
+
+
+## Conclusion
+
+To achieve a stable and resilient metric agent setup following configuration parameters added or adjusted.
+
+To avoid metric agent to overrun with huge amount of data floods from single scrape target and run in OOM, configuration parameter `sample_limit: 50000` added each prometheus receiver scrape configuration.
+The metric agent can handle higher values but analyzing current kyma production deployments shown values around `50000` in average.
+
+For more memory resilience, the processor `memory_limiter` configured with a hard limit of `75%`, hard and soft memory limit check intervals configured to `0.5 second` to allow processor react faster when limits are exceed.
+
+Batch processor configured with a batch size `1024` to avoid hit grpc client payload size limit during metric are exported to the gateway instances.
```

Comment:
```suggestion
Batch processor was configured with a batch size of `1024` to avoid hitting the gRPC client payload size limit while metrics are exported to the gateway instances.
```